v3/src/components/graph/adornments/adornment-models.ts

```diff
@@ -0,0 +1,74 @@
+/*
+  Adornment models are strictly MST. The keep track of the user modifications of the defaults.
+ */
+
+import {Instance, types} from "mobx-state-tree"
+import {uniqueId} from "../../../utilities/js-utils"
+import {Point} from "../utilities/graph_utils"
+
+export const PointModel = types.model("Point", {
+  x: types.optional( types.number,NaN),
+  y: types.optional( types.number,NaN)
+})
+  .actions(self => ({
+    set(aPt:Point | undefined) {
+      if(aPt) {
+        self.x = aPt.x
+        self.y = aPt.y
+      }
+    }
+  }))
+export interface IPointModel extends Instance<typeof PointModel> {}
+export const kInfinitePoint = {x:NaN, y:NaN}
+
+const Adornment = types.model("Adornment", {
+  id: types.optional(types.identifier, () => uniqueId()),
+  type: types.optional(types.string, () => {
+    throw "type must be overridden"
+  }),
+  isVisible: true
+})
+  .actions(self => ({
+    setVisibility(isVisible: boolean) {
+      self.isVisible = isVisible
+    }
+  }))
+
+export const MovableValueModel = Adornment
+  .named('MovableValueModel')
+  .props({
+    type: 'value',
+    value: types.number,
+  })
+  .actions(self => ({
+    setValue(aValue: number) {
+      self.value = aValue
+    }
+  }))
+export interface IMovableValueModel extends Instance<typeof MovableValueModel> {}
+
+export const MovableLineModel = Adornment
+  .named('MovableLineModel')
+  .props({
+    type: 'line',
+    intercept: types.number,
+    slope: types.number,
+    pivot1: types.optional( PointModel, kInfinitePoint),
+    pivot2: types.optional( PointModel, kInfinitePoint)
```

Comment:
Out of curiosity, do these need to be serialized or can they be determine at run time from other information?

Reply:
They do need to be serialized.